source/auth/tests/README.rst

```diff
@@ -0,0 +1,55 @@
+==========
+Auth Tests
+==========
+
+The YAML and JSON files in this directory tree are platform-independent tests
+that drivers can use to prove their conformance to the Auth Spec at least with
+respect to connection string URI input.
+
+Drivers should do additional unit testing if there are alternate ways of
+configuring credentials on a client.
+
+Driver must also conduct the prose tests in the Auth Spec test plan section.
+
+Format
+------
+
+Each YAML file contains an object with a single ``tests`` key. This key is an
+array of test case objects, each of which have the following keys:
+
+- ``description``: A string describing the test.
+- ``uri``: A string containing the URI to be parsed.
+- ``valid:`` A boolean indicating if the URI should be considered valid.
+- ``credential_configured:`` A boolean indicating if the URI is considered to have
+  configured a credential for the purposes of deciding if the driver should
+  authenticate to the topology.
+- ``credential``: An object containing the following keys:
+
+  - ``username``: A string containing the username. For auth mechanisms
+    that do not utilize a password, this may be the entire ``userinfo`` token
+    from the connection string.  If this is the empty string,
+    test harnesses should interpret it as unspecified.
+  - ``password``: A string containing the password.  If this is the empty string,
```

Comment:
I prefer null to the empty string for non-associating values.

Reply:
Yeah, it's clunky.  We already use null to mean "no assertion necessary" -- meaning any value or none at all is allowed -- so I was trying to succinctly say "required to be missing".   I'm open to other suggestions... false?  A separate "password_set" field?

Reply:
Where do we use `null` to mean any value or none at all?  In command monitoring expectations, we use an explicit `null` to mean that the value should not be present, which seems similar enough to this use case. In this context, it wouldn't make sense for `password: null` to mean "any value or none at all", so I'm fine with using `null` as the non-associating value here.  @jmikola any thoughts on the matter?

 @jmikola what do you think?

Reply:
Connection string tests say `If a test case includes a null value for one of these keys (e.g. auth: ~, port: ~), no assertion is necessary. ` and this was modeled on that and has a similar line.  

Another option would be to remove keys that don't need assertions (e.g. if valid if false, no need for a credentials field), and use null to mean a value that must not be set.  I was trying to deviate less from existing test runners, but maybe that was the wrong goal versus test clarity.